gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.ts

```diff
@@ -172,19 +186,50 @@ export class DocumentationNewPageComponent implements OnInit {
         },
       });
 
-    this.schema$ = this.fetcherService.getList().pipe(
-      map((list) => list.find((fetcher) => fetcher.id === this.httpFetcherName)?.schema),
-      map((schema) => JSON.parse(schema)),
-    );
+    this.fetcherService
+      .getList()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe(
+        (fetchers) =>
+          (this.fetchers = fetchers.map((f) => {
+            return {
+              id: f.id,
+              name: f.name,
+              schema: JSON.parse(f.schema),
+              iconPath: `assets/logo_${f.name.toLowerCase()}-fetcher.svg`,
+            };
+          })),
+      );
 
-    this.form.controls.source.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((value) => {
-      if (value === this.httpValue) {
+    this.form.controls.sourceType.valueChanges.pipe(distinctUntilChanged(), takeUntilDestroyed(this.destroyRef)).subscribe((value) => {
+      if (value === 'EXTERNAL') {
         this.form.controls.content.clearValidators();
+        this.form.controls.source.addValidators([Validators.required]);
+        this.sourceConfiguration?.addValidators([Validators.required]);
       } else {
         this.form.controls.content.addValidators([Validators.required]);
+        this.form.controls.source.clearValidators();
+        this.sourceConfiguration?.clearValidators();
       }
       this.form.controls.content.updateValueAndValidity();
+      this.form.controls.source.updateValueAndValidity();
+      this.sourceConfiguration?.updateValueAndValidity();
     });
+
+    this.form.controls.source.valueChanges
+      .pipe(
+        distinctUntilChanged(),
+        tap((_) => {
+          this.selectedFetcherSchema = undefined;
+          this.sourceConfiguration = undefined;
+        }),
+        debounceTime(500),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe((value) => {
+        this.sourceConfiguration = new FormControl<unknown>({}, [Validators.required]);
+        this.selectedFetcherSchema = this.fetchers.find((f) => f.id === value)?.schema;
+      });
```

Comment:
Do we need to add error?
Something like
`    error: (err) => {
      console.error('Error handling source value changes:', err);
    },`